File: py/inst/demoUserScript.py
import hal9 as h9
import pandas as pd

df = pd.DataFrame({'state': ("CA", "WA", "OR"),
    'statistic': (69, 420, 666)})
h9.set('df', df)

h9.dropdown('dropdown', values = lambda :h9.get('df').columns, on_update=lambda x: h9.set('value', x))

def filter_and_show_df(value):
    df = h9.get('df')
    return df[df['state'] == value].to_html()

h9.code('rawhtml', lambda :filter_and_show_df(h9.get('value')))

File: r/inst/demo-user-script.R
library(hal9)

iris |> h9_set("df")

h9_dropdown(
    "dropdown",
    values = c("setosa", "versicolor"),
    on_update = function(value) {
        h9_set(value, "selected_species")
    }
)

h9_code(
    "rawhtml",
    {
        df <- h9_get("df")
        selected_species <- h9_get("selected_species")
        df <- df[df$Species == selected_species, ]

        knitr::kable(df, format = "html") |>
            as.character()
    }
)
